posthog/api/organization.py

```diff
@@ -203,15 +207,41 @@ def create_team(self, organization: Organization, user: User) -> Team:
 
     def to_representation(self, instance) -> Dict:
         data = UserSerializer(instance=instance).data
-        data["redirect_url"] = "/personalization" if self.enable_new_onboarding() else "/ingestion"
+        data["redirect_url"] = "/personalization?new_account=true" if self.enable_new_onboarding() else "/ingestion"
```

Comment:
Can you clarify what/why is changing in the user flow? 

If I'm understanding things right:
1. We send new_account=true to FE which saves it as a flag somewhere
2. We read the flag in demo warnings code and flash it for a bit

What's the goal though? Can we do this in a more deterministic way which wouldn't break with reloads?

Reply:
Cool! So the intended behavior from a user perspective is:
- The first time I see the demo warning (just after creating my account), I want to see a different message (`welcome`).
- Any time after that, I want to see `incomplete_setup_on_demo_project` (assuming all the rest of the conditions to show such warnings are met).

I agree with your concern, this felt quite shaky, but also didn't feel like it was worth storing something on the BE for this. Maybe localStorage?

Reply:
:thinking: How about we show it if `organization.create_at` is less than 1 day ago (or until they reach next step)? This way we can keep the logic deterministic and isolated.

Reply:
Per discussion on Slack, this approach sounds good, updating now.